ts/src/base/Exchange.ts

```diff
@@ -6358,6 +6358,29 @@ export default class Exchange {
         return this.filterBySymbolSinceLimit (sorted, symbol, since, limit) as LongShortRatio[];
     }
 
+    handleTriggerDirectionAndParams (params, exchangeSpecificKey: string) {
+        /**
+         * @ignore
+         * @method
+         * @returns {[string, object]} the trigger-direction value and omited params
+         */
+        let triggerDirection = this.safeString (params, 'triggerDirection');
+        // throw exception if:
+        // A) if provided value is not unified (support old "up/down" strings too)
+        // B) if exchange specific "trigger direction key" (eg. "stopPriceSide") was not provided
+        if (!this.inArray (triggerDirection, [ 'ascending', 'descending', 'up', 'down' ]) && !(exchangeSpecificKey in params)) {
```

Comment:
instead of this complicated condition, why don't you return immediately if the exchangeSpecificKey is available

Reply:
actually this is in opposite, to avoid complication in implementations. in implementations we need the unified value, not exchange-specific value. 
```
[ triggerDirection, params ] = this.handle...
```
here triggerDirection (in implementations) we need to have either `asc/desc` or undefined, and we continue processing the request args depending that.
I am confident this is an optimal approach for now.
lmk your thoughts